.github/workflows/build.yml

```diff
@@ -0,0 +1,24 @@
+name: build
+
+on: [pull_request, push, workflow_dispatch]
+
+jobs:
+  build-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Pull Docker
+        run : docker pull thesofproject/sof && docker tag thesofproject/sof sof
+
+      - name: Init
+        run: echo -e '#!/bin/bash\nmkdir build && cd build && cmake .. && make -Werror -Wall -Wmissing-prototypes -Wimplicit-fallthrough=3 -Wpointer-arith' > build.sh && chmod +x build.sh
```

Comment:
Note you don't need to escape newlines in this case.

Alternatively, feel free to create a new `scripts/build-with-warnings.sh` file. Or later.